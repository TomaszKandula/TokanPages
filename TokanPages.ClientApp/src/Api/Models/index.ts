export type { AddArticleDto } from "./Articles/addArticleDto";
export type { UpdateArticleContentDto } from "./Articles/updateArticleContentDto";
export type { UpdateArticleCountDto } from "./Articles/updateArticleCountDto";
export type { UpdateArticleLikesDto } from "./Articles/updateArticleLikesDto";
export type { UpdateArticleVisibilityDto } from "./Articles/updateArticleVisibilityDto";
export type { RemoveArticleDto } from "./Articles/removeArticleDto";

export type { SendMessageDto } from "./Mailer/sendMessageDto";
export type { SendNewsletterDto } from "./Mailer/sendNewsletterDto";
export type { SubscriberInfoDto } from "./Mailer/subscriberInfoDto";
export type { VerifyEmailAddressDto } from "./Mailer/verifyEmailAddressDto";

export type { AddSubscriberDto } from "./Subscribers/addSubscriberDto";
export type { UpdateSubscriberDto } from "./Subscribers/updateSubscriberDto";
export type { RemoveSubscriberDto } from "./Subscribers/removeSubscriberDto";

export type { UserDataDto } from "./Users/userDataDto";
export type { ActivateUserDto } from "./Users/activateUserDto";
export type { AddUserDto } from "./Users/addUserDto";
export type { UpdateUserDto } from "./Users/updateUserDto";
export type { RemoveUserDto } from "./Users/removeUserDto";
export type { AuthenticateUserDto } from "./Users/authenticateUserDto";
export type { AuthenticateUserResultDto } from "./Users/authenticateUserResultDto";
export type { ReAuthenticateUserDto } from "./Users/reAuthenticateUserDto";
export type { ReAuthenticateUserResultDto } from "./Users/reAuthenticateUserResultDto";
export type { RevokeUserRefreshTokenDto } from "./Users/revokeUserRefreshTokenDto";
export type { ResetUserPasswordDto } from "./Users/resetUserPasswordDto";
export type { UpdateUserPasswordDto } from "./Users/updateUserPasswordDto";
export type { UserPermissionDto } from "./Users/userPermissionDto";
export type { UserRoleDto } from "./Users/userRoleDto";
export type { UploadUserMediaDto } from "./Users/uploadUserMediaDto";
export type { UploadUserMediaResultDto } from "./Users/uploadUserMediaResultDto";

export type { ErrorDto } from "./Error/errorDto";
export type { ValidationErrorsDto } from "./Error/validationErrorsDto";

export type { DocumentContentDto } from "./Components/documentContentDto";

export type { NavigationContentDto } from "./Components/navigationContentDto";
export type { HeaderContentDto } from "./Components/headerContentDto";
export type { FooterContentDto } from "./Components/footerContentDto";
export type { ClientsContentDto } from "./Components/clientsContentDto";
export type { ArticleFeaturesContentDto } from "./Components/articleFeaturesContentDto";
export type { ContactFormContentDto } from "./Components/contactFormContentDto";
export type { CookiesPromptContentDto } from "./Components/cookiesPromptContentDto";
export type { FeaturedContentDto } from "./Components/featuredContentDto";
export type { FeaturesContentDto } from "./Components/featuresContentDto";
export type { NewsletterContentDto } from "./Components/newsletterContentDto";
export type { ResetPasswordContentDto } from "./Components/resetPasswordContentDto";
export type { UpdatePasswordContentDto } from "./Components/updatePasswordContentDto";
export type { UserSigninContentDto } from "./Components/userSigninContentDto";
export type { UserSignoutContentDto } from "./Components/userSignoutContentDto";
export type { UserSignupContentDto } from "./Components/userSignupContentDto";
export type { TestimonialsContentDto } from "./Components/testimonialsContentDto";
export type { UnsubscribeContentDto } from "./Components/unsubscribeContentDto";
export type { ActivateAccountContentDto } from "./Components/activateAccountContentDto";
export type { UpdateSubscriberContentDto } from "./Components/updateSubscriberContentDto";
export type { WrongPagePromptContentDto } from "./Components/wrongPagePromptContentDto";
export type { 
    AccountContentDto, 
    SectionAccessDenied,
    SectionAccountInformation,
    SectionAccountPassword,
    SectionAccountRemoval
} from "./Components/accountContentDto";

export type { IconDto } from "./Components/Common/iconDto";
export type { LinkDto } from "./Components/Common/linkDto";
export type { RowItemDto } from "./Components/Common/rowItemDto";
export type { TextItemDto } from "./Components/Common/textItemDto";
export type { ContentDto } from "./Components/Common/contentDto";

export type { GetContentManifestDto } from "./Content/getContentManifestDto";
export type { LanguageItemDto } from "./Content/Items/languageItemDto";
export type { JWT } from "./Jwt";
