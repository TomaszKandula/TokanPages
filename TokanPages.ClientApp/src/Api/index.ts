export {
    IsSuccessStatusCode,
    GetBaseHeaders,
    GetProcessedBody,
    GetProcessedHeaders,
    GetProcessedResponse,
} from "./Utilities";

export type { ExecuteApiActionResultProps, ExecuteStoreActionProps, ExecuteApiActionProps } from "./Utilities/Types";

export {
    API_BASE_URI,
    GET_ARTICLES,
    GET_ARTICLE,
    GET_ARTICLE_BY_TITLE,
    ADD_ARTICLE,
    UPDATE_ARTICLE_CONTENT,
    UPDATE_ARTICLE_COUNT,
    UPDATE_ARTICLE_LIKES,
    UPDATE_ARTICLE_VISIBILITY,
    RETRIEVE_ARTICLE_INFO,
    REMOVE_ARTICLE,
    LOG_MESSAGE,
    GET_USERS,
    GET_USER,
    GET_USER_IMAGE,
    ADD_USER,
    ACTIVATE_USER,
    UPDATE_USER,
    REMOVE_USER,
    AUTHENTICATE,
    REAUTHENTICATE,
    VERIFY_USER_EMAIL,
    REVOKE_REFRESH_TOKEN,
    REVOKE_USER_TOKEN,
    RESET_USER_PASSWORD,
    UPDATE_USER_PASSWORD,
    UPLOAD_USER_IMAGE,
    GET_NEWSLETTERS,
    GET_NEWSLETTER,
    ADD_NEWSLETTER,
    UPDATE_NEWSLETTER,
    REMOVE_NEWSLETTER,
    SEND_MESSAGE,
    SEND_NEWSLETTER,
    NOTIFY_WEB_URL,
    NOTIFICATION_STATUS,
    GET_CONTENT_MANIFEST,
    REQUEST_PAGE_DATA,
    GET_ARTICLE_MAIN_IMAGE_URL,
    GET_IMAGES_URL,
    GET_DOCUMENTS_URL,
    ARTICLE_PATH,
    GET_USER_VIDEO,
    GET_VIDEO_ASSET,
    GET_USER_NOTES,
    GET_USER_NOTE,
    ADD_USER_NOTE,
    UPDATE_USER_NOTE,
    REMOVE_USER_NOTE,
} from "./Paths";
