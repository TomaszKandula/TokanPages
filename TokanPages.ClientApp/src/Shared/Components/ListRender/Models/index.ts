export type { Fields } from "./Fields";
export type { Item } from "./Item";
export type { Subitem } from "./Subitem";
