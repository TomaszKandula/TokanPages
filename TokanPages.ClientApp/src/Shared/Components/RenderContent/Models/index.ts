export type { IArticleItem } from "./ArticleItemModel";
export type { IAuthor } from "./AuthorModel";
export type { ITextObject, ITextItem, IRowItem } from "./TextModel";
