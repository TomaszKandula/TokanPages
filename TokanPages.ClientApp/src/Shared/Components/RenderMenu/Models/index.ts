export type { Fields, MenuProps } from "./Fields";
export type { Item } from "./Item";
export type { Subitem } from "./Subitem";
