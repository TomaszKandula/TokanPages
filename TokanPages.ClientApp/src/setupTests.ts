// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import enableHooks from "jest-react-hooks-shallow";

enableHooks(jest);

jest.mock("react-redux", () => (
{
    useSelector: jest.fn(),
    useDispatch: jest.fn()
}));

jest.mock("connected-react-router", () => (
{
    connectRouter: jest.fn(),
    routerMiddleware: jest.fn()
}));

Enzyme.configure({ adapter: new Adapter(), disableLifecycleMethods: false });

// environmental variables used during all tests
process.env.REACT_APP_API_VER="1";
process.env.REACT_APP_BACKEND="http://localhost:5000";
