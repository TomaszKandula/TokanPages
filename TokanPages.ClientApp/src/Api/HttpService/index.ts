import { ApplicationState } from "../../Store/Configuration";
import { USER_DATA } from "../../Shared/constants";
import { GetDataFromStorage } from "../../Shared/Services/StorageServices";
import { RaiseError } from "../../Shared/Services/ErrorServices";
import { AuthenticateUserResultDto } from "../Models";
import Validate from "validate.js";
import base64 from "base-64";
import utf8 from "utf8";

export interface HeadersProps {
    key: string;
    value: string;
}

export interface Configuration {
    method: string;
    body?: object;
    form?: FormData;
    headers?: HeadersProps[];
    hasJsonResponse: boolean;
}

export interface StoreProps {
    dispatch: (action: any) => void;
    state: () => ApplicationState;
}

export interface PromiseResult {
    status: number | null;
    content: any | null;
    error: any | null;
}

export interface GetContentRequest extends StoreProps {
    request: string;
    receive: string;
    url: string;
}

export interface ExecuteActionRequest {
    url: string;
    configuration: Configuration;
}

export interface ExecuteRequest extends ExecuteActionRequest, StoreProps {
    optionalHandle?: string;
    responseType?: string | string[];
}

const IsSuccessStatusCode = (statusCode: number): boolean => {
    return statusCode >= 200 && statusCode <= 299;
};

const SetupHeaders = (): HeadersProps[] => {
    const timezoneOffset = new Date().getTimezoneOffset();
    const encoded = GetDataFromStorage({ key: USER_DATA }) as string;

    const headers: HeadersProps[] = [];
    headers.push({ key: "UserTimezoneOffset", value: timezoneOffset.toString() });
    headers.push({ key: "Content-Type", value: "application/json" });

    if (Validate.isEmpty(encoded)) {
        return headers;
    }

    const decoded = base64.decode(encoded);
    const text = utf8.decode(decoded);
    const data = JSON.parse(text) as AuthenticateUserResultDto;
    const hasAuthorization = Validate.isObject(data) && !Validate.isEmpty(data.userToken);

    if (hasAuthorization) {
        headers.push({ key: "Authorization", value: `Bearer ${data.userToken}`});
    }

    return headers;
};

export const GetContentAction = (props: GetContentRequest): void => {
    let url = props.url;
    if (props.state !== undefined) {
        const id = props.state().applicationLanguage.id as string;
        const queryParam = Validate.isEmpty(id) ? "" : `&language=${id}`;
        url = `${props.url}${queryParam}`;
    }

    props.dispatch({ type: props.request });

    const input: ExecuteRequest = {
        dispatch: props.dispatch,
        state: props.state,
        url: url,
        responseType: props.receive,
        configuration: {
            method: "GET",
            headers: SetupHeaders(),
            hasJsonResponse: true,
        },
    };

    DispatchExecuteAction(input);
};

export const DispatchExecuteAction = async (props: ExecuteRequest): Promise<void> => {
    if (!props.dispatch || !props.state) {
        return;
    }

    const state = props.state();
    const components = state.contentPageData.components;
    const content = components.templates.templates.application;
    
    try {
        const optionalBody = props.configuration.body
        ? JSON.stringify(props.configuration.body)
        : null;

        const optionalFormData = props.configuration.form
        ? props.configuration.form
        : null;

        const body = optionalBody !== null 
        ? optionalBody 
        : optionalFormData !== null 
        ? optionalFormData 
        : null;

        const headers = new Headers();
        const baseHeaders = SetupHeaders();
        baseHeaders.forEach(header => {
            if (optionalFormData !== null && header.key === "Content-Type") {
            } else {
                headers.append(header.key, header.value);
            }
        });

        if (Array.isArray(props.configuration.headers)) {
            props.configuration.headers.forEach(header => {
                if (header.key === "Content-Type" && headers.has("Content-Type")) {
                    headers.delete("Content-Type");
                }

                headers.append(header.key, header.value);
            });
        }

        const response = await fetch(props.url, {
            method: props.configuration.method,
            headers: headers,
            body: body,
        });

        if (IsSuccessStatusCode(response.status)){
            let data: object | string = {};
            if (props.configuration.hasJsonResponse) {
                data = await response.json();
            } else {
                data = await response.text();
            }

            if (props.responseType !== undefined) {
                if (Array.isArray(props.responseType)) {
                    props.responseType.forEach(item => {
                        props.dispatch({ type: item, payload: data, handle: props.optionalHandle });
                    });
                } else {
                    props.dispatch({ type: props.responseType, payload: data, handle: props.optionalHandle });
                }
            } else {
                throw new Error(content.unexpectedError);
            }
        } else {
            const statusCode = response.status.toString();
            const statusText = content.unexpectedStatus.replace("{STATUS_CODE}", statusCode);
            throw new Error(statusText);
        }
    } catch (exception) {
        console.error(exception);
        RaiseError({
            dispatch: props.dispatch,
            errorObject: exception,
            content: content,
        });
    }
};

export const ExecuteAsync = async (props: ExecuteActionRequest): Promise<PromiseResult> => {
    let result: PromiseResult = { status: null, content: null, error: null };

    try {
        const optionalBody = props.configuration.body
        ? JSON.stringify(props.configuration.body)
        : null;

        const optionalFormData = props.configuration.form
        ? props.configuration.form
        : null;

        const body = optionalBody !== null 
        ? optionalBody 
        : optionalFormData !== null 
        ? optionalFormData 
        : null;

        const headers = new Headers();
        const baseHeaders = SetupHeaders();
        baseHeaders.forEach(header => {
            if (optionalFormData !== null && header.key === "Content-Type") {
            } else {
                headers.append(header.key, header.value);
            }
        });

        if (Array.isArray(props.configuration.headers)) {
            props.configuration.headers.forEach(header => {
                if (header.key === "Content-Type" && headers.has("Content-Type")) {
                    headers.delete("Content-Type");
                }

                headers.append(header.key, header.value);
            });
        }

        const response = await fetch(props.url, {
            method: props.configuration.method,
            headers: headers,
            body: body,
        });

        if (IsSuccessStatusCode(response.status)) {
            let data: object | string = {};
            if (props.configuration.hasJsonResponse) {
                data = await response.json();
            } else {
                data = await response.text();
            }

            result = {
                status: response.status,
                content: data,
                error: null,
            };
        } else {
            throw new Error("Unexpected error!"); 
        }
    } catch (exception) {
        result = {
            status: null,
            content: null,
            error: exception,
        };
    }

    return result;
};
