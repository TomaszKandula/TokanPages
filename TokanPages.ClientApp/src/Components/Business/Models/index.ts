import { ServiceItemDto, TechItemsDto } from "../../../Api/Models";
import { ReactChangeEvent, ReactMouseEvent } from "../../../Shared/types";

export interface MessageFormProps {
    company: string;
    firstName: string;
    lastName: string;
    email: string;
    phone: string;
    description: string;
    techStack: string[];
    services: string[];
}

export interface TechStackListProps {
    isDisabled: boolean;
    list: TechItemsDto[];
    handler: (event: ReactChangeEvent, isChecked: boolean) => void;
}

export interface ServiceItemCardProps {
    isDisabled: boolean;
    key: React.Key | null | undefined;
    value: ServiceItemDto;
    handler: (event: ReactMouseEvent) => void;
    services: string[];
}

export interface BusinessFormProps {
    hasCaption?: boolean;
    hasIcon?: boolean;
    hasShadow?: boolean;
    className?: string;
    background?: string;
}
