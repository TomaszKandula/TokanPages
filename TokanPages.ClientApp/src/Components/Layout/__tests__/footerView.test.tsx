import "../../../setupTests";
import React from "react";
import { shallow } from "enzyme";
import FooterView from "../footerView";
import { IFooterContentIconDto } from "../../../Api/Models";

describe("Test component: footerView.", () => 
{
    it("Renders correctly '<FooterView />' when content is loaded.", () => 
    {
        const icons: IFooterContentIconDto = 
        {
            name: "LinkedInIcon",
            link: "https://www.linkedin.com/"
        };

        const tree = shallow(<FooterView bind=
        {{
            terms: "Terms of use",
            policy: "Privacy policy",
            versionInfo: "1.0",
            hasVersionInfo: false,
            backgroundColor: "#FFFFFF",
            boxPaddingBottom: 0,
            copyright: "Â© 2020 - 2021 Tomasz Kandula",
            reserved: "All rights reserved",
            icons: [icons]
        }}/>);
        expect(tree).toMatchSnapshot();
    });
});
