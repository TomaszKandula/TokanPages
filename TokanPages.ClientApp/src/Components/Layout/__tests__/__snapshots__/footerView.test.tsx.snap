// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component: footerView. Renders correctly '<FooterView />' when content is loaded. 1`] = `
<footer
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Container))
    maxWidth="lg"
  >
    <div
      data-aos="zoom-in"
    >
      <Styled(MuiBox)
        alignItems="center"
        display="flex"
        flexWrap="wrap"
        pb={0}
        pt={6}
      >
        <WithStyles(ForwardRef(Typography))
          className="makeStyles-copy-4"
          component="p"
          gutterBottom={false}
        >
          Â© 2020 - 2021 Tomasz Kandula
           | 
          All rights reserved
           | 
          <SetTermsLink />
           | 
          <SetPolicyLink />
        </WithStyles(ForwardRef(Typography))>
        <Styled(MuiBox)
          className="makeStyles-icon_box-2"
          ml="auto"
        >
          <WithStyles(ForwardRef(IconButton))
            aria-label="LinkedInIcon"
            className="makeStyles-icon-3"
            color="default"
            href="https://www.linkedin.com/"
            key="0"
            target="_blank"
          >
            <Memo(ForwardRef(LinkedInIcon)) />
          </WithStyles(ForwardRef(IconButton))>
        </Styled(MuiBox)>
      </Styled(MuiBox)>
      <RenderVersionInfo />
    </div>
  </WithStyles(ForwardRef(Container))>
</footer>
`;
